src/ai/backend/cli/interaction.py

```diff
@@ -0,0 +1,116 @@
+import ipaddress
+import os
+
+
+def ask_host(prompt: str, default: str = "127.0.0.1") -> str:
+    try:
+        default = default.replace("localhost", "127.0.0.1")
+        _ = ipaddress.ip_address(default)
+    except ValueError:
+        raise ValueError("IP host must be given as 127.0.0.1")
+
+    while True:
+        user_reply = input(f"{prompt}(default: {default}): ")
+        if user_reply == "":
+            user_reply = default
+        try:
+            _ = ipaddress.ip_address(user_reply)
+            break
+        except ValueError:
+            print("Please input correct host.")
+    return user_reply
+
+
+def ask_number(prompt: str, default: int = 0, min_value: int = 0, max_value: int = 0) -> int:
```

Comment:
How about using `typing.TypeVar` to generalize `int` to any numeric type (e.g., float, int, decimal, etc.) specified the caller?